consensus/src/dag/adapter.rs

```diff
@@ -368,6 +385,22 @@ impl DAGStorage for StorageAdapter {
         // TODO: use callback from notifier to cache the latest ledger info
         Ok(self.aptos_db.get_latest_ledger_info()?)
     }
+
+    fn save_payload(&self, payload: &DecoupledPayload) -> anyhow::Result<()> {
+        Ok(self
+            .consensus_db
+            .put::<DecoupledPayloadSchema>(&payload.digest(), payload)?)
+    }
+
+    fn get_payloads(&self) -> anyhow::Result<Vec<(PayloadDigest, DecoupledPayload)>> {
```

Comment:
Don't you also need to get a single payload?

Reply:
We get everything to memory on start, so we don't need it yet.